site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,1032 @@
+:page-layout: default
+:page-title: integration-testing
+:page-permalink: /guides/integration-testing
+
+= Understand how to develop integration tests for your cloud-native Spring Boot applications
+
+An integral part of the lifecycle of any piece of production-grade software is proper application of testing methodologies. +
+Of the various types of tests that can be created, integration tests can often be some of the most difficult to create,
+not only because of the various real-world systems that need to be up and running (such as the PaaS, a database, other microservices etc), but also because
+the tooling isn't as well known as the tooling that exists for unit tests or tests that interact primarily with the framework the application is written in
+(think of the Spring Testing Framework here).
+
+Cloud-native applications are no exception. If anything the large number of interactions such applications may
+potentially have with other systems in the PaaS environment makes integration testing all the more necessary.
+
+This guide provides an in-depth analysis of how to create and configure your integration tests in order to properly test your
+Spring Boot application that is meant to run on Kubernetes or OpenShift.
```

Comment:
That's a good question.

The guide only touches on Openshift here. We could note that things may or may not work on Kubernetes but I'm not sure we want to make  in such an effort (since we would need to explain a host of different things).
@metacosm What have you done about this situation in the HTTP guide?

Reply:
I've only explained the differences when I had to i.e. when talking about the generated resources `Deployment` vs `DeploymentConfig` and the presence of `Route` in the OpenShift case… but I didn't go into details.

Reply:
Perhaps another guide then :wink: ? (with a focus on the developers perspective)

Reply:
That would be interesting since I don't even know what the exact differences are :)